pkg/dataingestionframework/data/dif_types.go

```diff
@@ -0,0 +1,52 @@
+package data
+
+import "fmt"
+
+//Data ingestion framework topology entity
+type DIFEntity struct {
+	UID  string `json:"uniqueId"`
+	Type string `json:"type"`
+	Name string `json:"name"`
+
+	HostedOn            *DIFHostedOn                 `json:"hostedOn"`
+	MatchingIdentifiers *DIFMatchingIdentifiers      `json:"matchIdentifiers"`
+	PartOf              []*DIFPartOf                 `json:"partOf"`
```

Comment:
Suggest adding builder function (e.g., `PartOf()` to make creating `PartOf` list easier. The function can check duplicates and avoid adding duplicated items.